Gemfile

```diff
@@ -0,0 +1,47 @@
+source 'https://rubygems.org'
+git_source(:github) { |repo| "https://github.com/#{repo}.git" }
+
+ruby '3.2.2'
+
+# Bundle edge Rails instead: gem "rails", github: "rails/rails", branch: "main"
+gem 'rails', '~> 7.0.8'
+
+# Use postgresql as the database for Active Record
+gem 'pg', '~> 1.1'
+
+# Use the Puma web server [https://github.com/puma/puma]
+gem 'puma', '~> 5.0'
+
+# Build JSON APIs with ease [https://github.com/rails/jbuilder]
+# gem "jbuilder"
+
+# Use Redis adapter to run Action Cable in production
+# gem "redis", "~> 4.0"
+
+# Use Kredis to get higher-level data types in Redis [https://github.com/rails/kredis]
+# gem "kredis"
+
+# Use Active Model has_secure_password [https://guides.rubyonrails.org/active_model_basics.html#securepassword]
+# gem "bcrypt", "~> 3.1.7"
+
+# Windows does not include zoneinfo files, so bundle the tzinfo-data gem
+gem 'tzinfo-data', platforms: %i[mingw mswin x64_mingw jruby]
+
+# Reduces boot times through caching; required in config/boot.rb
+gem 'bootsnap', require: false
+
+# Use Active Storage variants [https://guides.rubyonrails.org/active_storage_overview.html#transforming-images]
+# gem "image_processing", "~> 1.2"
+
+# Use Rack CORS for handling Cross-Origin Resource Sharing (CORS), making cross-origin AJAX possible
+# gem "rack-cors"
+
```

Comment:
- [ ] 🚀 Fantastic progress on your implementation! 🚀 However, I've noticed a minor hiccup that's preventing your front-end from displaying data from the backend, and it's a common culprit: the infamous `CORS` issue. Don't worry; Let's do this together🖐️.

Here's the lowdown:

**Issue:**
The browser console is throwing a CORS error, which is impeding communication between your front-end and back-end. See below for details of the error:
<img width="677" alt="Screenshot 2023-11-30 at 23 24 52" src="https://github.com/Gardimy/hello-rails-back-end/assets/107192076/dab5fb8c-5d64-4800-b3b7-6aea1bcb10df">


**Solution:**
Let's get that `CORS` issue sorted out by following these steps:

1. Uncomment the `rack-cors` gem in your gem file.
2. Locate the `cors.rb` file at `config/initializers/cors.rb` and uncomment the code from line 8 - 16.
3. In the `cors.rb` file, replace the value of `origins` with either the URL of the server or website that needs access to your API. Alternatively, you can use `*`, allowing anyone to access your API.

   ```ruby
       Rails.application.config.middleware.insert_before 0, Rack::Cors do
          allow do
            origins "example.com"

            resource "*",
               headers: :any,
               methods: [:get, :post, :put, :patch, :delete, :options, :head]
          end
      end
   ```

By ironing out these `CORS` wrinkles, we'll ensure smooth communication between your front-end and back-end. Happy coding! 🌟👨‍💻